docs/src/adoc/index.adoc

```diff
@@ -3780,41 +3780,123 @@ instance:
+  <artifactId>jdbi3-jackson2</artifactId>
+</dependency>
+----
 
 [source,java]
 ----
-// First, install the plugin and create a json column
 jdbi.installPlugin(new Jackson2Plugin());
-// You may need database support
-jdbi.installPlugin(new PostgresPlugin());
-jdbi.getConfig(Jackson2Config.class).setMapper(myObjectMapper); // optional
+// optional
+jdbi.getConfig(Jackson2Config.class).setMapper(myObjectMapper);
+----
+
+==== Gson 2
+
+This plugin provides JSON backing through Gson 2.
 
-handle.execute("create table jsons (id serial not null, value json not null)");
+[source,xml,subs="specialchars"]
+----
+<dependency>
+  <groupId>org.jdbi</groupId>
+  <artifactId>jdbi3-gson2</artifactId>
+</dependency>
+----
 
-class Whozit {} // any Jackson serializable type
+[source,java]
+----
+jdbi.installPlugin(new Gson2Plugin());
+// optional
+jdbi.getConfig(Gson2Config.class).setGson(myGson);
+----
 
-// Then, you can use @Json qualifying type:
-interface JsonDao {
-    @SqlUpdate("insert into jsons (json) values(:value)")
-    int insert(@Json Whozit value); // on parameters
+==== Operation
 
-    @SqlQuery("select value from jsons")
-    @Json // on result types
-    List<Whozit> select();
+Any bound object qualified as link:{jdbidocs}/json/Json.html[@Json^] — except `String` —
+will be converted by the link:{jdbidocs}/json/JsonConfig.html[registered^]
+link:{jdbidocs}/json/JsonMapper.html[JsonMapper^] and _requalified_ as `@Json String`.
+A correspondingly qualified `ArgumentFactory` will then be called to store the JSON data,
+allowing special JSON handling for your database to be implemented.
+If none are found, a factory for plain `String` will be used instead, to handle the JSON as plaintext.
+
+Mapping works just the same way, but in reverse: an output type qualified as `@Json T` will be fetched from a
+`@Json String` or `String` ColumnMapper, and then passed through the `JsonMapper`.
+
+[TIP]
+Our PostgresPlugin provides qualified factories that will
+bind/map the `@Json String` to/from `json` or `jsonb`-typed columns.
+
+==== Usage
+
+[source,java]
+----
+handle.execute("create table myjsons (id serial not null, value json not null)");
+----
+
+SqlObject:
+
+[source,java]
+----
+// any json-serializable type
+class MyJson {}
+
+// use @Json qualifier:
+interface MyJsonDao {
+    @SqlUpdate("insert into myjsons (json) values(:value)")
+    // on parameters
+    int insert(@Json MyJson value);
+
+    @SqlQuery("select value from myjsons")
+    // on result types
+    @Json
+    List<MyJson> select();
 }
 
-// It also works on bean or property-mapped properties:
+// also works on bean or property-mapped objects:
 class MyBean {
-    private final Whozit property;
+    private final MyJson property;
     @Json
-    public Whozit getProperty() { return ...; }
+    public MyJson getProperty() { return ...; }
 }
 ----
 
+Fluent API:
+
+[source,java]
+----
+QualifiedType qualifier = QualifiedType.of(MyJson.class).with(Json.class);
```

Comment:
This will need to become `QualifiedType<MyJson>` after #1396 merges

Reply:
The json doc is the first one to mention qualifiedtype so this will be the only one